lib/ddtrace/context.rb

```diff
@@ -203,25 +221,25 @@ def get
     # @return [Array<Span>] deleted spans
     def delete_span_if
       @mutex.synchronize do
-        [].tap do |deleted_spans|
-          @trace.delete_if do |span|
-            finished = span.finished?
+        deleted_spans = []
+        return deleted_spans unless block_given?
```

Comment:
This was a guard clause I added to avoid an error. Depends on how you look at it: `delete_span_if` without a block is basically saying delete nothing. So you could say "this is legal, even if useless" or "you probably made some kind of error, so I'll raise one."

All depends on how verbose we want to be here. I think we generally avoid raising errors in trace code like this because it tends to break tracing/add performance cost.

Reply:
I guess the question is: would we expect customers to use this method? If so, I think the clear semantics are very important -- we'd be helping them figure out that they actually have a bug in their code.

If it's just for us, it's just a minor "nice to have" point.